### PR TITLE
tests/ci: wait for terraform provide the ipxe and then start the nodes

### PR DESCRIPTION
**Pull request by cpanato**
cc @mxinden @s-urbaniak

### CODE REVIEW

File: tests/rspec/lib/metal_support.rb [hunk 1]
```diff
@@ -118,6 +117,29 @@ def self.wait_for_matchbox
...
+    fork do
+      from = Time.now
+      node_mac = varfile.tectonic_metal_controller_macs[0].tr(':', '-')
+      loop do
+        `curl --silent --fail -k \
+        "http://matchbox.example.com:8080/ipxe?uuid=&mac=#{node_mac}&domain=&hostname=&serial=" > /dev/null`
+        if $CHILD_STATUS.exitstatus != 0
+          puts "waiting for terraform. Last exit status: #{$CHILD_STATUS.exitstatus}"
+          elapsed = Time.now - from
+          raise 'matchbox never get a image to boot the nodes' if elapsed > 1200
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: s/get a/got an/

File: tests/rspec/lib/metal_support.rb [hunk 2]
```diff
@@ -118,6 +117,29 @@ def self.wait_for_matchbox
...
+          sleep 10
+        else
+          puts 'Terraform is ready'
+          puts 'Starting QEMU/KVM nodes'
+          system(env_variables_setup, 'sudo -E ./scripts/libvirt create')
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: `system` returns whether the command succeeded or not. Could you react on that exit code? Something like https://github.com/coreos/tectonic-installer/blob/master/tests/rspec/lib/cluster.rb#L170

File: tests/rspec/lib/metal_support.rb [hunk 3]
```diff
@@ -118,6 +117,29 @@ def self.wait_for_matchbox
     end
   end
 
+  def self.wait_for_terraform(varfile)
+    fork do
+      from = Time.now
+      node_mac = varfile.tectonic_metal_controller_macs[0].tr(':', '-')
+      loop do
+        `curl --silent --fail -k \
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: why using \` and not just `system` which returns the exit status right away?

File: tests/rspec/lib/metal_support.rb [hunk 4]
```diff
@@ -118,6 +117,29 @@ def self.wait_for_matchbox
     end
   end
 
+  def self.wait_for_terraform(varfile)
+    fork do
+      from = Time.now
+      node_mac = varfile.tectonic_metal_controller_macs[0].tr(':', '-')
+      loop do
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: How about using `with_retries` here? https://github.com/coreos/tectonic-installer/blob/master/tests/rspec/lib/with_retries.rb

File: tests/rspec/lib/metal_support.rb [hunk 5]
```diff
@@ -118,6 +117,31 @@ def self.wait_for_matchbox
     end
   end
 
+  # We fork this to run independently from the current execution because we need to wait for
+  # terraform apply reach some state that provide the ipxe to we boot the nodes.
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: s/apply reach/apply to reach/

s/to/before we/

Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: done

**Pull request merged**